### PR TITLE
Add instrumentation to monitor the impact of schema updates on compression efficiency

### PR DESCRIPTION
**Pull request by lquerel**
This PR is the first in a series of two PRs with the ultimate goal of eliminating the observed degradation of the compression rate over time for Lightstep's span-type data.
This first PR focuses on the instrumentation of the code to better understand the impact of certain schema changes on the average compression rate (see the chart below). 

image

The second PR will address this issue by identifying dictionary overflow that could be better handled as dictionary reset.

### CODE REVIEW

File: pkg/otel/common/schema/builder/duration.go [hunk 1]
```diff
@@ -54,10 +54,11 @@ func (b *DurationBuilder) Append(value arrow.Duration) {
 		return
 	}
 
-	if value != 0 {
+	if value != 0 && b.updateRequest != nil {
 		// If the builder is nil, then the transform node is not optional.
 		b.transformNode.RemoveOptional()
-		b.updateRequest.Inc()
+		b.updateRequest.Inc(&update.NewFieldEvent{FieldName: b.transformNode.Path()})
+		b.updateRequest = nil // No need to report this again.
```
Comment (jmacd) [inline pkg/otel/common/schema/builder/duration.go]: For the pattern here -- should the `b.updateRequest != nil` condition apply to lines 60-61 and not 59?
I.e., how come this is not
```
if value != 0 {
		// If the builder is nil, then the transform node is not optional.
		b.transformNode.RemoveOptional()
        if b.updateRequest != nil 
		        b.updateRequest.Inc(&update.NewFieldEvent{FieldName: b.transformNode.Path()})
		        b.updateRequest = nil 
       }
}
```

Comment (lquerel) [reply pkg/otel/common/schema/builder/duration.go]: It's intentional. In fact the previous version was not optimal and was doing the same idempotent node transformation again and again for each value added to the builder. This modification has the same effect on the tree of transformations while doing less modification on it.

**Review by jmacd — APPROVED** [review]

**Comment by jmacd** [comment]
I apparently closed this by pushing

**Comment by jmacd** [comment]
Opening again.

**Pull request closed without merging**